# HG changeset patch
# User Rik <octave@nomad.inbox5.com>
# Date 1274381382 25200
#      Thu May 20 11:49:42 2010 -0700
# Node ID 5c6b73a844e4e15b49ed0af7d4d46d6d43833166
# Parent  6c50b56824aa3aaff6dc1331a071eed4d5fb0913
Plot minor ticks for semilog plots

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,12 @@
+2010-05-16  Rik <octave@nomad.inbox5.com>
+
+        * plot/semilogx.m, plot/semilogy.m: Plot minor ticks for semilog plots.
+
 2010-05-16  Rik <octave@nomad.inbox5.com>
 
         * audio/loadaudio.m, audio/playaudio.m, audio/saveaudio.m,
         audio/wavread.m, general/bicubic.m, general/bitcmp.m, general/bitget.m,
         general/bitset.m, general/cplxpair.m, general/del2.m,
         general/gradient.m, general/interp2.m, general/interp3.m,
         general/interpn.m, general/postpad.m, general/prepad.m,
         general/private/__splinen__.m, image/image_viewer.m, image/imagesc.m,
diff --git a/scripts/plot/semilogx.m b/scripts/plot/semilogx.m
--- a/scripts/plot/semilogx.m
+++ b/scripts/plot/semilogx.m
@@ -32,16 +32,19 @@ function retval = semilogx (varargin)
   [h, varargin] = __plt_get_axis_arg__ ("semilogx", varargin{:});
 
   oldh = gca ();
   unwind_protect
     axes (h);
     newplot ();
 
     set (h, "xscale", "log");
+    if (any( strcmp (get (gca, "nextplot"), {"new", "replace"})))
+      set (h, "xminortick", "on");
+    endif
 
     tmp = __plt__ ("semilogx", h, varargin{:});
 
     if (nargout > 0)
       retval = tmp;
     endif
   unwind_protect_cleanup
     axes (oldh);
diff --git a/scripts/plot/semilogy.m b/scripts/plot/semilogy.m
--- a/scripts/plot/semilogy.m
+++ b/scripts/plot/semilogy.m
@@ -32,16 +32,19 @@ function retval = semilogy (varargin)
   [h, varargin] = __plt_get_axis_arg__ ("semilogy", varargin{:});
 
   oldh = gca ();
   unwind_protect
     axes (h);
     newplot ();
 
     set (h, "yscale", "log");
+    if (any( strcmp (get (gca, "nextplot"), {"new", "replace"})))
+      set (h, "yminortick", "on");
+    endif
 
     tmp = __plt__ ("semilogy", h, varargin{:});
 
     if (nargout > 0)
       retval = tmp;
     endif
 
   unwind_protect_cleanup
